packages/features/embed/types/index.d.ts

```diff
@@ -1,20 +1,40 @@
+import type { Brand } from "@calcom/types/utils";
+
 import type { tabs } from "../lib/EmbedTabs";
 import type { useEmbedTypes } from "../lib/hooks";
 
 export type EmbedType = "inline" | "floating-popup" | "element-click" | "email";
+type EmbedConfig = {
+  layout?: BookerLayouts;
+  theme?: Theme;
+};
+
 export type PreviewState = {
-  inline: {
-    width: string;
-    height: string;
-  };
+  inline: Brand<
+    {
+      width: string;
+      height: string;
+      config?: EmbedConfig;
+    },
+    "inline"
+  >;
```

Comment:
This guarantees that we don't accidentally mix one embedType's data with other as they might be structurally similar.